packages/flame/lib/src/geometry/ellipse_component.dart

```diff
@@ -0,0 +1,200 @@
+  /// Constructor to create an [EllipseComponent] from width and height.
+  EllipseComponent({
+    double? width,
+    double? height,
+    super.position,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.children,
+    super.priority,
+    super.paint,
+    super.paintLayers,
+    super.key,
+  }) : super(size: Vector2(width ?? 0, height ?? 0));
+
+  /// Constructor to create an [EllipseComponent] relative to the parent size.
+  /// The width and height are determined by
+  /// [widthRelation] and [heightRelation]
+  /// which are fractions of the parent's width and height.
+  EllipseComponent.relative(
+    double widthRelation,
+    double heightRelation, {
+    required Vector2 parentSize,
+    super.position,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.paint,
+    super.paintLayers,
+    super.children,
+    super.key,
+  }) : super(
+          size: Vector2(
+            widthRelation * parentSize.x,
+            heightRelation * parentSize.y,
+          ),
+        );
+
+  @override
+  @mustCallSuper
+  Future<void> onLoad() async {
+    void updateCenterOffset() => _centerOffset = Offset(size.x / 2, size.y / 2);
+    size.addListener(updateCenterOffset);
+    updateCenterOffset();
+  }
+
+  late Offset _centerOffset;
+
+  /// Get the width of the ellipse before scaling.
+  @override
+  double get width {
+    return size.x;
+  }
+
+  /// Set the width of the ellipse (and therefore the [size]).
+  @override
+  set width(double value) {
+    size.setValues(value, size.y);
+  }
+
+  /// Get the height of the ellipse before scaling.
+  @override
+  double get height {
+    return size.y;
+  }
+
+  /// Set the height of the ellipse (and therefore the [size]).
+  @override
+  set height(double value) {
+    size.setValues(size.x, value);
+  }
+
+  // Used to not create new Vector2 objects
+  // every time scaledWidth or scaledHeight is called.
+  final Vector2 _scaledSize = Vector2.zero();
+
+  /// Get the width of the ellipse after it has been sized and scaled.
+  double get scaledWidth {
+    _scaledSize
+      ..setFrom(size)
+      ..multiply(absoluteScale);
+    return _scaledSize.x;
+  }
+
+  /// Get the height of the ellipse after it has been sized and scaled.
+  double get scaledHeight {
+    _scaledSize
+      ..setFrom(size)
+      ..multiply(absoluteScale);
+    return _scaledSize.y;
+  }
+
+  @override
+  void render(Canvas canvas) {
+    if (renderShape) {
+      if (hasPaintLayers) {
+        for (final paint in paintLayers) {
+          canvas.drawOval(
+            Rect.fromCenter(
+              center: _centerOffset,
+              width: scaledWidth,
+              height: scaledHeight,
+            ),
+            paint,
+          );
+        }
+      } else {
+        canvas.drawOval(
+          Rect.fromCenter(
+            center: _centerOffset,
+            width: scaledWidth,
+            height: scaledHeight,
+          ),
```

Comment:
nit: you can extract the repeated rect definition to avoid duplication

Reply:
fixed